app/services/analysis/gwas/manhattan_plot.rb

```diff
@@ -1,58 +1,70 @@
-            trait = data_file.file_file_name.match(/SNPAssociation-Full-(.*)\.csv$/)[1]
+          neg_log10_p_values[name] = neg_log10_p
+        end
 
-            CSV.foreach(data_file.file.path) do |name, _, neg_log10_p|
-              # TODO: fix GWASSER so that it does not output invalid row ID.1
-              next if name =~ /\AID(\.1)?\z/
+        if map_csv_data_file = analysis.data_files.gwas_map.first
+          File.open(map_csv_data_file.file.path, "r") do |file|
+            map_data = Analysis::Gwas::MapCsvParser.new.call(file)
+            mutations = map_data.csv.map do |name, chrom, pos|
+              [name, neg_log10_p_values[name.gsub(/\W/, '_')].to_f, chrom, pos.to_i]
+            end
+          end
 
-              # GWASSER replaces dashes in mutation names with dots...
-              name = name.gsub(".", "-")
+          mutations = mutations.sort do |mut_a, mut_b|
+            chrom_a, pos_a = mut_a[-2..-1]
+            chrom_b, pos_b = mut_b[-2..-1]
 
-              neg_log10_p_values[name] = neg_log10_p
+            if chrom_a.match(/\d/) && chrom_b.match(/\d/)
+              chrom_a = chrom_a.gsub(/\D+/, '').to_i
+              chrom_b = chrom_b.gsub(/\D+/, '').to_i
             end
 
-            if map_csv_data_file = analysis.data_files.gwas_map.first
-              File.open(map_csv_data_file.file.path, "r") do |file|
-                map_data = Analysis::Gwas::MapCsvParser.new.call(file)
-                mutations = map_data.csv.map do |name, chrom, pos|
-                  [name, neg_log10_p_values[name], chrom.to_i, pos.to_i]
-                end
-              end
-
-              mutations = mutations.sort do |mut_a, mut_b|
-                chrom_a, pos_a = mut_a[-2..-1]
-                chrom_b, pos_b = mut_b[-2..-1]
-
-                chrom_a == chrom_b ? pos_a <=> pos_b : chrom_a <=> chrom_b
-              end
-
-              mutations.each_with_index do |(_, _, chrom, _), idx|
-                chromosomes[chrom] << idx
-              end
-            else
-              mutations = neg_log10_p_values.to_a
-            end
+            chrom_a == chrom_b ? pos_a <=> pos_b : chrom_a <=> chrom_b
+          end
+
+        else
+          mutations = neg_log10_p_values.transform_values { |value| value.to_f }.to_a
+        end
+
+        mutations.reject! { |_, neg_log10_p| neg_log10_p < cutoff } if cutoff > 0
+
+        tooltips = mutations.map { |mut| format_tooltip(*mut) }
+
+        [trait, mutations, tooltips]
+      end
+
+      def call
+        { traits: [], chromosomes: [], cutoff: cutoff }.tap do |results|
+
+          chromosomes = Hash.new { |h, k| h[k] = [] }
+
+          result_data_files.each do |data_file|
+            trait, mutations, tooltips = process_trait(data_file)
 
-            tooltips = mutations.map { |mut| format_tooltip(*mut) }
+            mutations.each_with_index do |(_, _, chrom, _), idx|
+              chromosomes[chrom] << idx
+            end if mutations.first.size > 2
```

Comment:
I tried to run GWAS with a very simple input (2 PSUs, 1 trait, generated geno and map files) and I get

```undefined method `size' for nil:NilClass```

 in this line when trying to see the results. The analysis finished with success.